src/main/java/org/springframework/data/elasticsearch/core/index/MappingParameters.java

```diff
@@ -136,6 +139,9 @@ private MappingParameters(Field field) {
 		similarity = field.similarity();
 		termVector = field.termVector();
 		scalingFactor = field.scalingFactor();
+		maxShingleSize = field.maxShingleSize();
```

Comment:
```suggestion
		maxShingleSize = field.maxShingleSize() >= 0 ? field.maxShingleSize : null;
```
The check in the Assert should account for a not set default value of null